4-js/index.js

```diff
@@ -0,0 +1,330 @@
+  }
+}
+
+function createTdElement(value, parent) {
+  const td = document.createElement('td');
+  td.innerText = value;
+  parent.appendChild(td);
+}
+
+function createTdButtonElement(value, parent, style) {
+  const button = document.createElement('button');
+  const td = document.createElement('td');
+  button.name = value;
+  button.classList.add(style);
+  td.appendChild(button);
+  parent.appendChild(td);
+  if (style === 'btn-edit') {
+    button.addEventListener('click', showUserForm);
+  } else {
+    button.addEventListener('click', checkCheckbox);
+  }
+}
+
+function checkCheckbox(el) {
+  if (
+    el.currentTarget.parentNode.parentNode.children[0].children[0].children[0]
+      .checked
+  ) {
+    deleteUser(
+      el.currentTarget.parentNode.parentNode.children[0].children[0].children[0]
+    );
+  } else {
+    alert('The checkbox should be highlighted!');
+  }
+}
+
+function deleteUser(el) {
+  sendRequsest('DELETE', requestUrl + '/' + el.name)
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+  el.parentNode.parentNode.parentNode.remove();
+}
+
+function showUserForm(el) {
+  formUser.classList.remove('is-hide');
+  editUserBtn.classList.remove('is-hide');
+  titleEditUser.classList.remove('is-hide');
+  getEditObj(el.currentTarget.name);
+}
+
+function getEditObj(id) {
+    console.log(id);
+  sendRequsest('GET', requestUrl + '/' + id)
+    .then((data) => addValueToForm(data))
+    .catch((err) => console.log(err));
+}
+
+function addValueToForm(obj) {
+  userCompany.value = obj.company;
+  userCompany.name = obj.id;
+  userName.value = obj.name;
+  userAddress.value = obj.address;
+  userCity.value = obj.city;
+  userCountry.value = obj.country;
+}
+
+function editUser() {
+  let newUser = {
+    id: userCompany.name,
+    address: userAddress.value,
+    city: userCity.value,
+    company: userCompany.value,
+    name: userName.value,
+    country: userCountry.value,
+  };
+  sendRequsest('PUT', requestUrl + '/' + newUser.id, newUser)
+    .then((data) => renderEditUser(data))
+    .catch((err) => console.log(err));
+}
+
+function deleteSelectedUsers() {
+  let arrDeleteUsers = [];
+  for (let i = 0; i < arrUserCheckbox.length; i++) {
+    if (arrUserCheckbox[i].checked) {
+      arrDeleteUsers.push(arrUserCheckbox[i]);
+    }
+  }
+  for (let j = 0; j < arrDeleteUsers.length; j++) {
+    deleteUser(arrDeleteUsers[j]);
+  }
+}
+
+function renderEditUser(obj) {
+  const el = document.getElementById(obj.id);
+  el.children[1].innerText = obj.company;
+  el.children[2].innerText = obj.name;
+  el.children[3].innerText = obj.address;
+  el.children[4].innerText = obj.city;
+  el.children[5].innerText = obj.country;
+}
+
+function sendRequsest(method, url, body) {
+  const headers = {
+    'Content-Type': 'application/json',
+  };
+  return fetch(url, {
+    method: method,
+    body: JSON.stringify(body),
+    headers: headers,
+  }).then((response) => {
+    if (response.ok) {
```

Comment:
I would use a stotus instead of `ok` value

Reply:
changed